### PR TITLE
Feature/media cwt module

### PR DESCRIPTION
**Pull request by sverekar**
Edge worker compatible CWT module. 

**Limitations**
- Currently the module only support API's for verification of CWT tokens that are generated using HS256, ES256 algorithm only.
- Currently the module only support MAC0 COSE , COSE SIGN1 message structure. Refer this [page](https://datatracker.ietf.org/doc/rfc8152/) for more details on CBOR Object Signing and Encryption (COSE).
- As of now, EW do not support KMI to manage verification keys, hence these keys are fetched from property manager user defined variable which might not be a secure way. More details on user defined variables can be found [here](https://techdocs.akamai.com/property-mgr/docs/user-defined-vars)

### Dependencies/Todo
- [ ] E2E tests for ES256 CWT tokens.

### CODE REVIEW

File: delivery/common/cwt/lib/cwt.js [hunk 1]
```diff
@@ -0,0 +1,1216 @@
...
+                    return (128 & o) > 0 ? void (position$1 -= 15) : fromCharCode(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o);
+                }
+            }
+        }
+    }
+}
+
+let f32Array = new Float32Array(1), u8Array = new Uint8Array(f32Array.buffer, 0, 4);
+
+new Array(4096);
```
Comment (evan-hughes) [inline delivery/common/cwt/lib/cwt.js:472]: This doesn't seem to be used. Is it necessary?

**Review by evan-hughes — APPROVED** [review]
The portions of code that appear to be Akamai-authored meet our guidelines.

**Comment by sverekar** [comment]
Hi @evan-hughes , @bmatthew 

I have added support for ES256 algorithm for CWT module. Kindly review and provide your valuable feedback. 
Also, function definition have not been changed. 

Regards,

File: delivery/common/cwt/examples/typescript/src/cwt/cwt.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+}, HeaderLabelToKey_alg = 1, HeaderLabelToKey_crit = 2, claimsLabelToKey_iss = 1, claimsLabelToKey_sub = 2, claimsLabelToKey_aud = 3, claimsLabelToKey_exp = 4, claimsLabelToKey_nbf = 5;
+
+class Mac {
+    static async verifyHMAC(alg, message, signature, keys) {
+        if ("HMAC 256/256" === alg) {
+            let isSignVerified = !1;
+            for (const key of keys) if (isSignVerified = await crypto.subtle.verify({
+                name: "HMAC"
+            }, key, signature, message), isSignVerified) return Promise.resolve(isSignVerified);
+            return Promise.resolve(isSignVerified);
```
Comment (evan-hughes) [inline delivery/common/cwt/examples/typescript/src/cwt/cwt.js:18]: The function is async, so I don't think it's necessary to create a resolved Promise.

**Review by evan-hughes — APPROVED** [review]
Looks fine. It may be possible to simplify the code (very) slightly by dropping the calls to `Promise.resolve()` for the return values of async functions, but that's not significant.

**Review by evan-hughes — APPROVED** [review]
Looks fine. It may be possible to simplify the code (very) slightly by dropping the calls to `Promise.resolve()` for the return values of async functions, but that's not significant.

**Pull request merged**